great_expectations/datasource/data_connector/file_path_data_connector.py

```diff
@@ -324,7 +325,7 @@ def _map_batch_definition_to_data_reference(
         group_names: List[str] = regex_config["group_names"]
         return map_batch_definition_to_data_reference_string_using_regex(
             batch_definition=batch_definition,
-            regex_pattern=pattern,
+            regex_pattern=re.compile(pattern),
```

Comment:
This change to V3 usage is done so that the same utility method can support both, V3 as well as ZEP (where paths are handled more formally).